gadget/gadget.go

```diff
@@ -581,6 +591,14 @@ func InfoFromGadgetYaml(gadgetYaml []byte, model Model) (*Info, error) {
 		}
 	}
 
+	// Check if the allow values are valid kernel arguments
+	for _, arg := range gi.KernelCmdline.Allow {
+		parsed := osutil.ParseKernelCommandline(arg)
```

Comment:
don't we also need to check globbing, i.e. that if a param has `*` in it, it's all there is?

Reply:
my point is should we support `foo=b*` ? right now we take as a literal `b*` but that might be confusing going forward

Reply:
We actually already have that problem as we cannot tell the difference between `*` for globing or `*` as a literal. To do this properly we need to be able to escape and quote in a shell-like way. Which maybe we should do, and use the kernel parsing for the options and assume shell globing expressions for the allow list in the gadget. But that can make writing the allow list maybe less intuitive as things won't be exactly the same as what you would see in the kernel command line if there are globing characters involved. On the other hand I don't see *, ?, [, ] being used too much in the kernel command line so maybe it is fine requesting escaping in those corner cases.

But maybe what we can do is what I think you suggest and for the moment ban setting option values that can be interpreted as globing expressions.